### PR TITLE
Macepa lite tf - Adding MACEPA Lite dashboard 

### PR DESCRIPTION
**Pull request by justinlewis**
We tried to follow the existing coding practices as much as possible.

### CODE REVIEW

**Comment by justinlewis** [comment]
Please let me know what your procedure is for getting that snapshot failure resolved on Circle CI. I'm not sure how to proceed.

**Comment by justinlewis** [comment]
@moshthepitt - We believe we have adequate coverage in general. Regarding your comment below I followed the existing example when creating that snapshot exception and am not able to look into how React Jest snapshotting works in more detail. Sorry we are out of time for modifying the tests. 

> @justinlewis is it possible to avoid this if we follow the ideas here: https://kentcdodds.com/blog/effective-snapshot-testing/
> 
> The overall general intention is to slowly remove all the large snapshots, so I am quite uneasy about adding new ones.

**Comment by jsmethie** [comment]
In terms of .env values for staging:
```
REACT_APP_SUPERSET_IRS_LITE_REPORTING_JURISDICTIONS_DATA_SLICES=625
REACT_APP_SUPERSET_IRS_LITE_REPORTING_JURISDICTIONS_FOCUS_AREA_LEVEL=3
REACT_APP_SUPERSET_IRS_LITE_REPORTING_JURISDICTIONS_FOCUS_AREA_SLICE=617
REACT_APP_SUPERSET_IRS_LITE_REPORTING_JURISDICTIONS_COLUMNS=irsLiteZambiaJurisdictions2020

#columns for Zambia IRS Lite jurisdictions districts and Catchment Area
REACT_APP_SUPERSET_IRS_LITE_REPORTING_FOCUS_AREAS_COLUMNS=irsLiteZambiaFocusArea2020
REACT_APP_SUPERSET_IRS_LITE_REPORTING_INDICATOR_STOPS=zambia2020
REACT_APP_SUPERSET_IRS_LITE_REPORTING_INDICATOR_ROWS=zambia2020

# Irs Lite reporting config identifiers - these will change to select the correct configs for the client
REACT_APP_SUPERSET_IRS_LITE_REPORTING_JURISDICTIONS_DATA_SLICE=625
REACT_APP_SUPERSET_IRS_LITE_REPORTING_STRUCTURES_DATA_SLICE=620
REACT_APP_SUPERSET_IRS_LITE_REPORTING_PLANS_SLICE=621

# Irs Lite Reporting Config keys - these will never change, we will just keep adding them for different clients
REACT_APP_SUPERSET_IRS_LITE_REPORTING_JURISDICTIONS_DATA_SLICE_NA=0
REACT_APP_SUPERSET_IRS_LITE_REPORTING_STRUCTURES_DATA_SLICE_NA=0
REACT_APP_SUPERSET_IRS_LITE_REPORTING_PLANS_SLICE_NA=0

```
And for Prod:
```
REACT_APP_SUPERSET_IRS_LITE_REPORTING_JURISDICTIONS_DATA_SLICES=20
REACT_APP_SUPERSET_IRS_LITE_REPORTING_JURISDICTIONS_FOCUS_AREA_LEVEL=3
REACT_APP_SUPERSET_IRS_LITE_REPORTING_JURISDICTIONS_FOCUS_AREA_SLICE=19
REACT_APP_SUPERSET_IRS_LITE_REPORTING_JURISDICTIONS_COLUMNS=irsLiteZambiaJurisdictions2020

#columns for Zambia IRS Lite jurisdictions districts and Catchment Area
REACT_APP_SUPERSET_IRS_LITE_REPORTING_FOCUS_AREAS_COLUMNS=irsLiteZambiaFocusArea2020
REACT_APP_SUPERSET_IRS_LITE_REPORTING_INDICATOR_STOPS=zambia2020
REACT_APP_SUPERSET_IRS_LITE_REPORTING_INDICATOR_ROWS=zambia2020

# Irs Lite reporting config identifiers - these will change to select the correct configs for the client
REACT_APP_SUPERSET_IRS_LITE_REPORTING_JURISDICTIONS_DATA_SLICE=20
REACT_APP_SUPERSET_IRS_LITE_REPORTING_STRUCTURES_DATA_SLICE=21
REACT_APP_SUPERSET_IRS_LITE_REPORTING_PLANS_SLICE=18

# Irs Lite Reporting Config keys - these will never change, we will just keep adding them for different clients
REACT_APP_SUPERSET_IRS_LITE_REPORTING_JURISDICTIONS_DATA_SLICE_NA=0
REACT_APP_SUPERSET_IRS_LITE_REPORTING_STRUCTURES_DATA_SLICE_NA=0
REACT_APP_SUPERSET_IRS_LITE_REPORTING_PLANS_SLICE_NA=0

```

**Review by moshthepitt — CHANGES_REQUESTED** [review]
@justinlewis 

- can we make the tests pass please?
- there seems to be a conflict here `src/configs/env.ts` - can you fix this please?

**Comment by ciremusyoka** [comment]
@justinlewis Can we have the added ENVs documented. Check [here](https://github.com/onaio/reveal-frontend/blob/macepa-lite-tf/docs/env.md).

File: src/configs/settings.ts [hunk 1]
```diff
@@ -1729,6 +1756,26 @@ export const irsReportingCongif: {
...
+  irsLite2020: indicatorThresholdsIRSLite,
+};
+
+/* tslint:disable:object-literal-sort-keys */
+/** The actual configuration object controlling how IRS Reporting is handled for different clients */
+export const irsLiteReportingCongif: {
+  [key: string]: IrsReportingConfig;
+} = {
+  // Namibia Structures Configs
+  [process.env.REACT_APP_SUPERSET_IRS_LITE_REPORTING_STRUCTURES_DATA_SLICE_NA as string]: {
```
Comment (ciremusyoka) [inline src/configs/settings.ts:1773]: Lets follow the convention we already have. 
Move the processing of envs to `env.ts` file and also provide a default value in cases were the env is not provided.

File: src/containers/pages/IRSLite/MapLite/index.tsx [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+        </Col>
+      </Row>
+      <Row noGutters={true}>
+        <Col xs={9}>
+          {gisidaWrapperProps ? (
+            <div className="map irs-reporting-map">
+              <GisidaWrapper {...gisidaWrapperProps} />
+            </div>
+          ) : (
+            <div>{MAP_LOAD_ERROR}</div>
```
Comment (ciremusyoka) [inline src/containers/pages/IRSLite/MapLite/index.tsx:302]: We did away with `GisidaWrapper` in favour of [GisidaLite](https://github.com/onaio/reveal-frontend/blob/master/src/components/GisidaLite/index.tsx).

Comment (ciremusyoka) [reply src/containers/pages/IRSLite/MapLite/index.tsx:302]: This also affects the current `src/containers/pages/IRSLite/MapLite/helpers.ts` code

File: src/containers/pages/IRSLite/MapLite/helpers.ts [hunk 3]
```diff
@@ -0,0 +1,306 @@
+import GeojsonExtent from '@mapbox/geojson-extent';
```
Comment (ciremusyoka) [inline src/containers/pages/IRSLite/MapLite/helpers.ts:1]: Most of the code here is duplicated from [IRS map helpers](https://github.com/onaio/reveal-frontend/blob/macepa-lite-tf/src/containers/pages/IRS/Map/helpers.ts) why not import?

File: src/containers/pages/IRSLite/MapLite/index.tsx (hunk 2 above)
Comment (justinlewis) [reply src/containers/pages/IRSLite/MapLite/index.tsx]: This was converted to use GisdaLite by trying to reuse buildGsLiteLayers() in the FocusInvestigation component. Please see that method to make sure it aligns with your recent refactor.

File: src/configs/settings.ts (hunk 1 above)
Comment (justinlewis) [reply src/configs/settings.ts:1779]: This is following the conventions in the code. Look two methods higher to see the existing irsReportingConfig that this was coppied from.

**Comment by justinlewis** [comment]
> @justinlewis
> 
> * can we make the tests pass please?
> * there seems to be a conflict here `src/configs/env.ts` - can you fix this please?

@moshthepitt We believe we have adequate test coverage now. All the tests in IRSLite as well as any external test we could find related to the changes we made are passing.

File: src/configs/env.ts [hunk 4]
```diff
@@ -536,14 +536,13 @@ export const SUPERSET_IRS_MOPUP_REPORTING_JURISDICTIONS_FOCUS_AREA_LEVEL =
...
-  process.env.REACT_APP_SUPERSET_IRS_LITE_MOPUP_REPORTING_JURISDICTIONS_DATA_SLICES ?? '0';
-export type SUPERSET_IRS_LITE_MOPUP_REPORTING_JURISDICTIONS_DATA_SLICES = typeof SUPERSET_IRS_LITE_MOPUP_REPORTING_JURISDICTIONS_DATA_SLICES;
-
-export const SUPERSET_IRS_LITE_MOPUP_REPORTING_JURISDICTIONS_FOCUS_AREA_LEVEL =
-  process.env.REACT_APP_SUPERSET_IRS_LITE_MOPUP_REPORTING_JURISDICTIONS_FOCUS_AREA_LEVEL ?? '0';
-export type SUPERSET_IRS_LITE_MOPUP_REPORTING_JURISDICTIONS_FOCUS_AREA_LEVEL = typeof SUPERSET_IRS_LITE_MOPUP_REPORTING_JURISDICTIONS_FOCUS_AREA_LEVEL;
+export const ASSIGN_TEAMS_PLAN_TYPES_DISPLAYED = (
+  process.env.REACT_APP_ASSIGN_TEAMS_PLAN_TYPES_DISPLAYED ||
+  process.env.REACT_APP_DISPLAYED_PLAN_TYPES ||
+  'FI,IRS,MDA,MDA-Point,Dynamic-FI,Dynamic-IRS,Dynamic-MDA, IRS-Lite'
```
Comment (moshthepitt) [inline src/configs/env.ts]: Lets remove the whitespace before IRS-Lite

File: src/components/forms/PlanForm/tests/__snapshots__/index.test.tsx.snap [hunk 5]
```diff
@@ -68,7 +68,7 @@ exports[`containers/forms/PlanForm - Edit ACTIVE renders all fields correctly in
   readOnly={false}
```
Comment (moshthepitt) [inline src/components/forms/PlanForm/tests/__snapshots__/index.test.tsx.snap:68]: @justinlewis why does this PR result in changes to this file?  It should not be touching this code, no?

File: src/configs/env.ts [hunk 6]
```diff
@@ -457,6 +502,22 @@ export const SUPERSET_IRS_SOP_BY_DATE_PERFORMANCE_REPORT_SLICE =
...
+export type SUPERSET_IRS_LITE_DATA_COLLECTORS_PERFORMANCE_REPORT_SLICE = typeof SUPERSET_IRS_LITE_DATA_COLLECTORS_PERFORMANCE_REPORT_SLICE;
+
+export const SUPERSET_IRS_LITE_SOP_PERFORMANCE_REPORT_SLICE =
+  process.env.REACT_APP_SUPERSET_IRS_LITE_SOP_PERFORMANCE_REPORT_SLICE || '0';
+export type SUPERSET_IRS_LITE_SOP_PERFORMANCE_REPORT_SLICE = typeof SUPERSET_IRS_LITE_SOP_PERFORMANCE_REPORT_SLICE;
+
+export const SUPERSET_IRS_LITE_SOP_BY_DATE_PERFORMANCE_REPORT_SLICE =
+  process.env.REACT_APP_SUPERSET_IRS_LITE_SOP_BY_DATE_PERFORMANCE_REPORT_SLICE || '0';
+export type SUPERSET_IRS_LITE_SOP_BY_DATE_PERFORMANCE_REPORT_SLICE = typeof SUPERSET_IRS_LITE_SOP_BY_DATE_PERFORMANCE_REPORT_SLICE;
+
```
Comment (moshthepitt) [inline src/configs/env.ts:520]: Do we have IRS-Lite performance reports?  If not then we should get rid of these.

**Review by moshthepitt — CHANGES_REQUESTED** [review]

File: src/configs/env.ts [hunk 7]
```diff
@@ -80,6 +84,11 @@ export const ENABLE_IRS_PERFORMANCE_REPORT =
   process.env.REACT_APP_ENABLE_IRS_PERFORMANCE_REPORT === 'true';
 export type ENABLE_IRS_PERFORMANCE_REPORT = typeof ENABLE_IRS_PERFORMANCE_REPORT;
 
+/** Do you want to enable IRS Lite Performance Reporting */
+export const ENABLE_IRS_LITE_PERFORMANCE_REPORT =
+  process.env.REACT_APP_ENABLE_IRS_LITE_PERFORMANCE_REPORT === 'true';
+export type ENABLE_IRS_LITE_PERFORMANCE_REPORT = typeof ENABLE_IRS_LITE_PERFORMANCE_REPORT;
+
```
Comment (ciremusyoka) [inline src/configs/env.ts:91]: This may not be needed since we don't have IRS lite performance report. Right ?

File: src/containers/pages/IRSLite/MapLite/helpers.ts [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+      value: 'spraysuccess',
+    },
+  ],
+};
+
+/** Utility function to build structure layer
+ * @param {StructureFeatureCollection} structures - Feature Collection of structures
+ * @param {string[][]} indicatorStops - the indicator stops
+ */
+export const structuresLayerBuilder = (
```
Comment (ciremusyoka) [inline src/containers/pages/IRSLite/MapLite/helpers.ts:95]: This function should be removed. It is no longer used after doing away with `GisidaWrapper`.

File: src/containers/pages/IRSLite/MapLite/helpers.ts (hunk 3 above)
Comment (ciremusyoka) [reply src/containers/pages/IRSLite/MapLite/helpers.ts]: @justinlewis this is not yet addressed.

File: src/containers/pages/FocusInvestigation/map/active/helpers/utils.tsx [hunk 9]
```diff
@@ -304,8 +304,15 @@ export interface CircleColor {
   stops: string[][];
   type: string;
 }
+export interface PolygonColor {
+  property: string;
+  stops: string[][];
+  type: string;
+}
```
Comment (ciremusyoka) [inline src/containers/pages/FocusInvestigation/map/active/helpers/utils.tsx:311]: tiny pick: this interface is similar to the one above it `CircleColor`.

**Comment by ciremusyoka** [comment]
@justinlewis  I used the above given ENVs and the map seems to be off. Anything i maybe missing?
irs-lite-map

cc @moshthepitt

File: src/containers/pages/IRSLite/MapLite/helpers.ts [hunk 10]
```diff
@@ -0,0 +1,269 @@
...
+  ['Not Sprayed', TASK_RED],
+  ['Partially Sprayed', TASK_ORANGE],
+  ['Not Visited', TASK_YELLOW],
+  ['Not Eligible', BLACK],
+];
+
+/** IRS Indicator stops
+ * These are all the indicator stops for IRS that we know about.
+ */
+export const IRSIndicatorStops: { [key: string]: string[][] } = {
```
Comment (ciremusyoka) [inline src/containers/pages/IRSLite/MapLite/helpers.ts:23]: tiny pick: rename this const to 'IRSLiteIndicatorStops'

**Comment by jsmethie** [comment]
> @justinlewis I used the above given ENVs and the map seems to be off. Anything i maybe missing?
> irs-lite-map
>
> cc @moshthepitt

No, there is nothing off in the map.  The Sompani jurisdiction that was created actually has a structure in Seattle, Washington.  So the map is zoomed out by deafult to include that.

File: src/configs/env.ts (hunk 4 above)
Comment (justinlewis) [reply src/configs/env.ts]: done

File: src/configs/env.ts (hunk 6 above)
Comment (justinlewis) [reply src/configs/env.ts:520]: Done

File: src/configs/env.ts (hunk 7 above)
Comment (justinlewis) [reply src/configs/env.ts:91]: Done

File: src/containers/pages/IRSLite/MapLite/helpers.ts (hunk 8 above)
Comment (justinlewis) [reply src/containers/pages/IRSLite/MapLite/helpers.ts:95]: Done

File: src/containers/pages/FocusInvestigation/map/active/helpers/utils.tsx (hunk 9 above)
Comment (justinlewis) [reply src/containers/pages/FocusInvestigation/map/active/helpers/utils.tsx:311]: Ya. I personally think there should be a GenericColor interface but since the existing CircleColor was there I thought the best practice would be to be excplicit (e.g. Polygon...).

File: src/configs/env.ts (hunk 4 above)
Comment (moshthepitt) [reply src/configs/env.ts]: This was because this is being evaluated by splitting on comma so the whitespace would affect things and we'd end up with a value that has an unexpected space and wouldn't match the plan intervention type

File: src/containers/pages/IRSLite/MapLite/helpers.ts (hunk 10 above)
Comment (justinlewis) [reply src/containers/pages/IRSLite/MapLite/helpers.ts:23]: I just removed this entirely in favor of just  utilizing default:

/** The default indicator stop */
export const defaultIndicatorStop = [
  ['Sprayed', TASK_GREEN],
  ['Not Sprayed', TASK_RED],
  ['Partially Sprayed', TASK_ORANGE],
  ['Not Visited', TASK_YELLOW],
  ['Not Eligible', BLACK],
];

/** IRS Lite Indicator stops
 * These are all the indicator stops for IRS Lite that we know about.
 */
export const IRSLiteIndicatorStops: { [key: string]: string[][] } = {
  zambia2020: defaultIndicatorStop,
};

File: src/components/formatting/ResponseAdherence.tsx [hunk 11]
```diff
@@ -18,7 +18,7 @@ const ResponseAdherence: React.ElementType = () => {
             <tbody>
```
Comment (moshthepitt) [inline src/components/formatting/ResponseAdherence.tsx:18]: @justinlewis @jsmethie could we also revert the changes to this file since they are unrelated?

File: src/containers/pages/InterventionPlan/NewPlan/General/index.tsx [hunk 12]
```diff
@@ -73,6 +73,17 @@ const planFormPropsLookUp = {
...
+    allowMoreJurisdictions: true,
+    cascadingSelect: true,
+    initialValues: {
+      ...defaultInitialValues,
+      activities: planActivitiesMap[InterventionType.IRS],
+      interventionType: InterventionType.IRS,
+    },
+    jurisdictionLabel: FOCUS_AREA_HEADER,
+    redirectAfterAction: PLAN_LIST_URL,
+  },
```
Comment (moshthepitt) [inline src/containers/pages/InterventionPlan/NewPlan/General/index.tsx:86]: @justinlewis @jsmethie let's please revert this one as well.  We are not yet adding IRSLite to `NewPlan`

File: src/components/formatting/ResponseAdherence.tsx (hunk 11 above)
Comment (justinlewis) [reply src/components/formatting/ResponseAdherence.tsx:18]: I reverted this but note that the change was because the HTML wasn't rendering correctly. One of you might want to change thse:

                <td>>=</td>
to

                <td>;&gt;=</td>

File: src/containers/pages/InterventionPlan/NewPlan/General/index.tsx (hunk 12 above)
Comment (justinlewis) [reply src/containers/pages/InterventionPlan/NewPlan/General/index.tsx:86]: Done

File: src/configs/env.ts (hunk 4 above)
Comment (justinlewis) [reply src/configs/env.ts]: Gotcha. It's fixed now so shouldn't be a problem.

**Review by ciremusyoka — APPROVED** [review]

**Review by moshthepitt — APPROVED** [review]
